python/pyarrow/array.pxi

```diff
@@ -1566,6 +1568,9 @@ cdef class Array(_PandasConvertible):
         -------
         array : numpy.ndarray
         """
+        if "numpy" not in sys.modules:
+            raise ValueError(
+                "Cannot return a numpy.ndarray if Numpy is not present")
```

Comment:
What should be our policy for errors here? `ValueError` could also be raised for other reasons. Perhaps `ImportError` would be more idiomatic? @jorisvandenbossche